packages/sales-api-service/src/services/recurring-payments.service.js

```diff
@@ -0,0 +1,54 @@
+import { findByExample, findById, Permission, RecurringPayment } from '@defra-fish/dynamics-lib'
+
+export const getRecurringPayments = async () => {
+  const currentDate = new Date()
+  const dueDates = []
+  for (let i = 0; i <= 10; i += 2) {
+    const date = new Date(currentDate)
+    date.setDate(currentDate.getDate() - i)
+    dueDates.push(date)
+  }
+
+  const recurringPayments = []
+  for (const dueDate of dueDates) {
+    const activeRecurringPayments = await findByExample(new RecurringPayment())
+    const dueRecurringPayments = await findRecurringPaymentsInDateRange(activeRecurringPayments, dueDate)
+
+    if (dueRecurringPayments.length !== 0) {
+      recurringPayments.push(dueRecurringPayments)
+    }
+  }
+
+  return recurringPayments
+}
+
+export const retrieveActivePermission = async recurringPayments => {
+  const recurringPaymentsWithPermission = []
+  for (const recurringPayment of recurringPayments) {
+    const permission = await findById(Permission, recurringPayment.activePermission)
+    recurringPayment.activePermission = permission
+    recurringPaymentsWithPermission.push(recurringPayment)
+  }
+  return recurringPaymentsWithPermission
+}
```

Comment:
This isn't how you should retrieve the related permission. If you look in dynamics-lib/src/queries/permission.queries.js, you'll see that it retrieves related entities for the Permission: Contact, Permit and Concessions. You should be doing it in the same way.